### PR TITLE
WFLY-11563 Race condition in PersistenceUnitServiceImpl when the service is being stopped

### PR DESCRIPTION
**Pull request by scottmarlow**
https://issues.jboss.org/browse/WFLY-11563

Also needed to update ClassFileTransformerTestCase proxy class (org.jboss.as.test.integration.jpa.mockprovider.classtransformer.TestEntityManagerFactory) to ignore the "isOpen" method call.

### CODE REVIEW

File: jpa/subsystem/src/main/java/org/jboss/as/jpa/service/PersistenceUnitServiceImpl.java [hunk 1]
```diff
@@ -248,20 +248,26 @@ public Void run() {
                                 if(javaNamespaceSetup != null) {
                                     javaNamespaceSetup.setup(Collections.<String, Object>emptyMap());
                                 }
+                                final EntityManagerFactory theEntityManagerFactory = entityManagerFactory;
```
Comment (jamezp) [inline jpa/subsystem/src/main/java/org/jboss/as/jpa/service/PersistenceUnitServiceImpl.java]: If you're using the double check locking idiom this won't be required.

File: jpa/subsystem/src/main/java/org/jboss/as/jpa/service/PersistenceUnitServiceImpl.java [hunk 2]
```diff
@@ -270,9 +276,14 @@ public Void run() {
                                         javaNamespaceSetup.teardown(Collections.<String, Object>emptyMap());
                                     }
                                 }
-                                if (proxyBeanManager != null) {
-                                    proxyBeanManager.setDelegate(null);
-                                    proxyBeanManager = null;
+                                final ProxyBeanManager theProxyBeanManager = proxyBeanManager;
```
Comment (jamezp) [inline jpa/subsystem/src/main/java/org/jboss/as/jpa/service/PersistenceUnitServiceImpl.java]: If you're using the double check locking idiom this won't be required.

**Review by jamezp — CHANGES_REQUESTED** [review]

File: jpa/subsystem/src/main/java/org/jboss/as/jpa/service/PersistenceUnitServiceImpl.java (hunk 2 above)
Comment (scottmarlow) [reply jpa/subsystem/src/main/java/org/jboss/as/jpa/service/PersistenceUnitServiceImpl.java]: good point

**Review by jamezp — APPROVED** [review]

**Comment by scottmarlow** [comment]
I'm not sure of what the "all checks have failed" message is about, https://ci.wildfly.org/viewLog.html?buildId=135272&buildTypeId=WF_PullRequest_WindowsJdk11 shows 2 failures and other tests passed.

**Pull request merged**